emulator/CMakeLists.txt

```diff
@@ -0,0 +1,44 @@
+set(EMULATOR_COMMON_SRCS
+    riscv_config.h
+    riscv_platform.c
+    riscv_platform.h
+    riscv_platform_impl.c
+    riscv_platform_impl.h
+    riscv_prelude.c
+    riscv_prelude.h
+    riscv_sail.h
+    riscv_sim.c
+    riscv_softfloat.c
+    riscv_softfloat.h
+)
+
+foreach (arch IN LISTS ENABLED_ARCHITECTURES)
+    add_executable(emulator_${arch}
+        "${CMAKE_BINARY_DIR}/riscv_model_${arch}.c"
+        ${EMULATOR_COMMON_SRCS}
+    )
+
+    add_dependencies(emulator_${arch} generated_model_${arch})
+
+    target_link_libraries(emulator_${arch}
+        PRIVATE softfloat sail_runtime GMP::GMP ZLIB::ZLIB
+    )
+
+    target_include_directories(emulator_${arch}
+        # So the generated C can find riscv_platform/prelude.h"
+        PRIVATE "${CMAKE_CURRENT_SOURCE_DIR}"
+    )
```

Comment:
Another option would be to set [`CMAKE_INCLUDE_CURRENT_DIR`](https://cmake.org/cmake/help/latest/variable/CMAKE_INCLUDE_CURRENT_DIR.html) to add this globally

Reply:
Ah didn't know about that. Updated.

Reply:
Actually I couldn't get this to work for some reason, even if I put it in `emulator/CMakeLists.txt`.

Reply:
Odd but explicit include dirs are probably better anyway.